src/SourceBuild/content/eng/finish-source-only.proj

```diff
@@ -172,4 +175,78 @@
+    <ItemGroup>
+      <ArtifactsPackageToBundle Include="$(ArtifactsShippingPackagesDir)**;
+                                          $(ArtifactsNonShippingPackagesDir)**"
+                                Condition="!$([System.String]::Copy('%(Identity)').EndsWith('.symbols.nupkg'))" />
+      <ReferencePackageToBundle Include="$(ReferencePackagesDir)**"
+                                Condition="!$([System.String]::Copy('%(Identity)').EndsWith('.symbols.nupkg'))" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <!-- Create a layout directory for the files that are to be included in the artifacts tarball. -->
+      <SourceBuiltLayoutDir>$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)', 'artifacts-layout'))</SourceBuiltLayoutDir>
+
+      <!-- Outputs -->
+      <SourceBuiltTarballName>$(ArtifactsAssetsDir)$(SourceBuiltArtifactsTarballName).$(SourceBuiltSdkVersion).$(TargetRid)$(ArchiveExtension)</SourceBuiltTarballName>
+      <SourceBuiltVersionName>$(SourceBuiltLayoutDir).version</SourceBuiltVersionName>
+      <AllPackageVersionsPropsName>$(SourceBuiltLayoutDir)PackageVersions.props</AllPackageVersionsPropsName>
+      <SourceBuiltMergedAssetManifestName>$(SourceBuiltLayoutDir)%(MergedAssetManifest.Filename)%(MergedAssetManifest.Extension)</SourceBuiltMergedAssetManifestName>
+    </PropertyGroup>
+  </Target>
+
+  <!-- Create the SourceBuilt.Private.Artifacts archive when building source-only. -->
+  <UsingTask TaskName="Microsoft.DotNet.UnifiedBuild.Tasks.WritePackageVersionsProps" AssemblyFile="$(MicrosoftDotNetUnifiedBuildTasksAssembly)" TaskFactory="TaskHostFactory" />
+  <Target Name="CreatePrivateSourceBuiltArtifactsArchive"
+          AfterTargets="Build"
+          DependsOnTargets="GetInputsOutputForCreatePrivateSourceBuiltArtifactsArchive"
+          Inputs="@(ArtifactsPackageToBundle);@(ReferencePackageToBundle);$(MergedAssetManifestOutputPath)"
+          Outputs="$(SourceBuiltTarballName);$(SourceBuiltVersionName);$(AllPackageVersionsPropsName);$(SourceBuiltMergedAssetManifestName)">
+    <!-- Copy packages to layout directory. Since there are a large number of files,
+          this will use symlinks instead of copying files to make this execute quickly. -->
+    <Copy SourceFiles="@(ArtifactsPackageToBundle)"
+          DestinationFolder="$(SourceBuiltLayoutDir)"
+          UseSymbolicLinksIfPossible="true" />
+    <Copy SourceFiles="@(ReferencePackageToBundle)"
+          DestinationFolder="$(SourceBuiltLayoutDir)SourceBuildReferencePackages"
+          UseSymbolicLinksIfPossible="true" />
+
+    <!-- Content of the .version file to include in the tarball -->
+    <ItemGroup>
+      <VersionFileContent Include="$(RepositoryCommit);$(SourceBuiltSdkVersion)" />
+    </ItemGroup>
+
+    <WriteLinesToFile File="$(SourceBuiltVersionName)"
+                      Lines="@(VersionFileContent)"
+                      Overwrite="true" />
+
+    <!-- Copy the merged asset manifest into the tarball -->
```

Comment:
Do you see the manifest replacing the PackageVersions.props entirely at some point?  Or another way, do you see the properties removed from the PackageVersions.props that come from the merge manifest?

Reply:
I could definitely see us only using the manifest. We'd have to make some changes to the bootstrapping project to handle the ExtraPackageVersionPropsPackageInfo items below itself to avoid breaks.

I have another branch that moves the `WritePackageVersionProps` below to be based off the vertical manifest and changes the consumption of the previously-source-built assets (when generating Versions.props) to use the asset manifest (which obviously needs to wait for the next rebaselining).